### PR TITLE
3.0/feature/iframe

### PR DESCRIPTION
**Pull request by annalinneajohansson**
Iframe component with GDPR adjustments.

### CODE REVIEW

File: source/php/Component/Iframe/Iframe.php [hunk 1]
```diff
@@ -2,15 +2,104 @@
...
+        }
+
+        if (isset($loading)) {
+            $this->data['attributeList']['loading'] = $loading;
+        }
+
+        $this->data['attributeList']['src'] = "about:blank";
+
+        if (isset($src)) {
+            $src_parsed = parse_url($src);
```
Comment (sebastianthulin) [inline source/php/Component/Iframe/Iframe.php:32]: Camelcase

File: source/php/Component/Iframe/Iframe.php [hunk 2]
```diff
@@ -2,15 +2,104 @@
...
+
+            $embed_url = "//{$src_parsed['host']}";
```
Comment (sebastianthulin) [inline source/php/Component/Iframe/Iframe.php:34]: Camelcase

File: source/php/Component/Iframe/Iframe.php [hunk 3]
```diff
@@ -2,15 +2,104 @@
...
+                        if (isset($supplier->policy)) {
+                            $this->data['attributeList']['data-supplier-policy'] = $supplier->policy;
+                        }
+                    }
+                }
+            }
         }
-	}
+    }
+    public function getSuppliers()
```
Comment (sebastianthulin) [inline source/php/Component/Iframe/Iframe.php:56]: Static?

File: source/php/Component/Iframe/Iframe.php [hunk 4]
```diff
@@ -2,15 +2,104 @@
...
+
+        if (isset($src)) {
+            $src_parsed = parse_url($src);
+
+            $embed_url = "//{$src_parsed['host']}";
+            $embed_url .= isset($src_parsed['path']) ? $src_parsed['path'] : '';
+
+            $this->data['attributeList']['data-src'] = $embed_url;
+
+            $suppliers = $this->getSuppliers();
```
Comment (sebastianthulin) [inline source/php/Component/Iframe/Iframe.php:39]: Static?

File: source/php/Component/Iframe/iframe.blade.php [hunk 5]
```diff
@@ -2,10 +2,6 @@
 <iframe 
     id="{{ $id }}" 
     class="{{ $class }}" 
-    data-src="{{ $src }}" 
-    loading="{{ $loading }}" 
-    height="{{ $height }}" 
-    width="{{ $width }}"
-    frameborder="0" 
-    src="{{ $src }}">
+	options="{{ $options }}"
```
Comment (sebastianthulin) [inline source/php/Component/Iframe/iframe.blade.php:5]: Not a valid attribute?

Comment (annalinneajohansson) [reply source/php/Component/Iframe/iframe.blade.php:5]: Having issues with double quotation marks in the json string that $options contains being formatted incorrectly when being parsed by [buildAttributes](https://github.com/helsingborg-stad/component-library/blob/b613c0b6923a67f638b4bc581b6cf2be0c19136f/source/php/Component/BaseController.php#L301-L313) so I put it in like this temporarily.

File: source/php/Component/Iframe/Iframe.php [hunk 6]
```diff
@@ -2,15 +2,104 @@
...
+
+            if (is_array($suppliers)) {
```
[the remark this answers is not in the document]
Comment (annalinneajohansson) [reply source/php/Component/Iframe/Iframe.php]: https://github.com/helsingborg-stad/component-library/pull/120/commits/f868c61a3050e84d91b4869eba09004b9750f3be

File: source/php/Component/Iframe/Iframe.php [hunk 7]
```diff
@@ -2,15 +2,107 @@
...
+                'KommersAnnons.se',
+                array( 'kommersannons.se' ),
+                'https://kommersannons.se/'
+            ),
+
+        );
+
+        return $suppliers;
+    }
+    private function setSupplierDataAttributes(string $host, $data)
```
Comment (annalinneajohansson) [inline source/php/Component/Iframe/Iframe.php:76]: Exception handling needed

File: source/php/Component/Iframe/Iframe.php [hunk 8]
```diff
@@ -4,114 +4,113 @@
...
+            new Supplier(
+                'KommersAnnons.se',
+                array( 'kommersannons.se' ),
+                'https://kommersannons.se/'
+            ),
+
+        );
+
+        if (function_exists('apply_filters')) {
+            $suppliers = apply_filters(__NAMESPACE__ . '\\' . ucfirst(__FUNCTION__), $suppliers);
```
Comment (sebastianthulin) [inline source/php/Component/Iframe/Iframe.php:79]: May be a candidate for a return here? We always strive to return as soon as possible.

File: source/php/Component/Iframe/Iframe.php (hunk 6 above)
[the remark this answers is not in the document]
Comment (annalinneajohansson) [reply source/php/Component/Iframe/Iframe.php]: @sebastianthulin Is this correct? https://github.com/helsingborg-stad/component-library/blob/3d99f733c6e76acb802a0ca5c3b6c8f217e468d3/source/php/Component/Iframe/Iframe.php#L39

**Comment by annalinneajohansson** [comment]
Dependant on: 

- https://github.com/helsingborg-stad/Municipio/pull/335
- https://github.com/helsingborg-stad/styleguide/pull/543

File: source/php/Component/Iframe/iframe.json [hunk 9]
```diff
@@ -1,15 +1,18 @@
...
+		"loading":"lazy",
+		"width":"100%",
+		"height": "400",
+		"frameborder":"0",
+		"options":""
+	},
+	"view":"iframe.blade.php",
+	"dependency": {
+		"sass": {
+			"components": [
```
Comment (sebastianthulin) [inline source/php/Component/Iframe/iframe.json:14]: Typography, button, etc. What other componets are used in this component?

**Pull request merged**